pytest_plugins/issue_handlers.py

```diff
@@ -0,0 +1,294 @@
+        issue = f"{match[0]}:{match[1]}"
+        if validation(data, issue, usage, **kwargs):
+            data[issue.strip()]['used_in'].append({'usage': usage, **kwargs})
+
+
+def _should_deselect(issue, data=None):
+    """Check if test should be deselected based on marked issue."""
+    # Handlers can be extended to support different issue trackers.
+    handlers = {'BZ': bugzilla.should_deselect_bz}
+    supported_handlers = tuple(f"{handler}:" for handler in handlers.keys())
+    if str(issue).startswith(supported_handlers):
+        handler_code, _, _ = str(issue).partition(":")
+        return handlers[handler_code.strip()](issue.strip(), data)
+
+
+def generate_issue_collection(items, config):  # pragma: no cover
+    """Generates a dictionary with the usage of Issue blockers
+
+    For use in pytest_collection_modifyitems hook
+
+    Arguments:
+        items {list} - List of pytest test case objects.
+        config {dict} - Pytest config object.
+
+    Returns:
+        [List of dicts] - Dicts indexed by "<handler>:<issue>"
+
+        Example of return data::
+
+            {
+                "XX:1625783" {
+                    "data": {
+                        # data taken from REST api,
+                        "status": ...,
+                        "resolution": ...,
+                        ...
+                        # Calculated data
+                        "is_open": bool,
+                        "is_deselected": bool,
+                        "clones": [list],
+                        "dupe_data": {dict}
+                    },
+                    "used_in" [
+                        {
+                            "filepath": "tests/foreman/ui/test_sync.py",
+                            "lineno": 124,
+                            "testcase": "test_positive_sync_custom_ostree_repo",
+                            "component": "Repositories",
+                            "usage": "skip_if_open"
+                        },
+                        ...
+                    ]
+                },
+                ...
+            }
+    """
+    settings.configure()
+    valid_markers = ["skip_if_open", "skip", "deselect"]
+    collected_data = defaultdict(lambda: {"data": {}, "used_in": []})
+
+    bz_cache_filename = config.getvalue('bz_cache')  # use existing json cache?
+    cached_data = None
+    if bz_cache_filename is not None:
+        try:
+            with open(bz_cache_filename, 'r') as bz_cache_file:
+                LOGGER.info(f'Using BZ cache file for issue collection: {bz_cache_filename}')
+                cached_data = {
+                    k: search_version_key(k, v) for k, v in json.load(bz_cache_file).items()
+                }
+        except FileNotFoundError:
+            # no bz cache file exists
+            LOGGER.warning(
+                f'bz-cache option used, file [{bz_cache_filename}] not found, '
+                f'NOT using cached data.'
+            )
+            pass
```

Comment:
could use pathlib here and check that the file exists first.
```python
bz_cache = Path(bz_cache_filename)
if bz_cache.exists():
    with bz_cache.open() as bz_cache_file:
        LOGGER.info(f'Using BZ cache file for issue collection: {bz_cache_filename}')
        cached_data = {
            k: search_version_key(k, v) for k, v in json.load(bz_cache_file).items()
        }
else:
    LOGGER.warning(
        f'bz-cache option used, file [{bz_cache_filename}] not found, NOT using cached data.'
    )
```

Reply:
Optional suggestion or would you rather see that pattern?  I prefer the try/except pattern in cases like this, maybe catching more than `FileNotFoundError` (like `OSError`) would make this more robust as it would account for many other file failures.

Reply:
do we expect an OSError? as a general rule, i try not to use try's! :D

Reply:
well, the file could exist, but not be readable, or writeable, etc - all of which fall under OSError.